src/internal/operations.rs

```diff
@@ -179,7 +179,7 @@ impl WriteEvents {
             Ok(()) as std::io::Result<()>
         });
 
-        promise.await.unwrap()
+        promise.await.map_err(|_| types::OperationError::ConnectionHasDropped)?
```

Comment:
`ConnectionHasDropped` has a different meaning internally, In this situation, a better error would be `ConnectionClosed`. That enum variant doesn't exist yet and I suggest to add it.